android/app/src/main/kotlin/com/jabook/app/jabook/audio/ExoPlayerMainThreadGuard.kt

```
@@ -0,0 +1,142 @@
+     * @param timeoutMs Maximum wait time in milliseconds (default: 5s)
+     * @param block Lambda receiving the player, executed on main thread
+     * @return Result of [block], or null if timed out or interrupted
+     */
+    fun <T> runOnMain(
+        timeoutMs: Long = DEFAULT_TIMEOUT_MS,
+        block: Player.() -> T,
+    ): T? {
+        if (Looper.myLooper() == Looper.getMainLooper()) {
+            return try {
+                player.block()
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)